<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServerWidget</class>
 <widget class="QWidget" name="ServerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>776</width>
    <height>538</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabPosition">
      <enum>QTabWidget::West</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_status">
      <attribute name="icon">
       <iconset theme="utilities-system-monitor">
        <normaloff/>
       </iconset>
      </attribute>
      <attribute name="title">
       <string>Status</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QTableView" name="table_server_status">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QAbstractScrollArea::AdjustToContents</enum>
           </property>
           <attribute name="horizontalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
          </widget>
         </item>
         <item>
          <widget class="QTableWidget" name="tableWidget">
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>64</number>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Address</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Ping</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>PL</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Score</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Edict</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Kick</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Ban</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Mute</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="button_refresh_status">
           <property name="text">
            <string>&amp;Refresh</string>
           </property>
           <property name="icon">
            <iconset theme="view-refresh">
             <normaloff/>
            </iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Last Updated</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_refresh_status">
           <property name="text">
            <string>(time)</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_console">
      <attribute name="icon">
       <iconset theme="utilities-terminal">
        <normaloff/>
       </iconset>
      </attribute>
      <attribute name="title">
       <string>Console</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="HistoryLineEdit" name="input_console"/>
         </item>
         <item>
          <widget class="QPushButton" name="button_send">
           <property name="text">
            <string>Sen&amp;d</string>
           </property>
           <property name="icon">
            <iconset theme="system-run">
             <normaloff/>
            </iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="output_console">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::NoWrap</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_cvars">
      <attribute name="icon">
       <iconset theme="format-list-unordered"/>
      </attribute>
      <attribute name="title">
       <string>Cvars</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QTableView" name="table_cvars"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="button_refresh_cvars">
           <property name="text">
            <string>&amp;Refresh</string>
           </property>
           <property name="icon">
            <iconset theme="view-refresh">
             <normaloff/>
            </iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Last Updated</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_refresh_cvar">
           <property name="text">
            <string>(time)</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_connection"/>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="button_reconnect">
       <property name="text">
        <string>Reconnect</string>
       </property>
       <property name="icon">
        <iconset theme="network-connect">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_setup">
       <property name="text">
        <string>&amp;Connection...</string>
       </property>
       <property name="icon">
        <iconset theme="configure">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <action name="action_clear_log">
   <property name="icon">
    <iconset theme="edit-clear">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Clear</string>
   </property>
  </action>
  <action name="action_attach_log">
   <property name="icon">
    <iconset theme="network-connect">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Attach Log</string>
   </property>
  </action>
  <action name="action_detach_log">
   <property name="icon">
    <iconset theme="network-disconnect">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Detach Log</string>
   </property>
  </action>
  <action name="action_parse_colors">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="format-text-color">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Parse Colors</string>
   </property>
  </action>
  <action name="action_save_log">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Save...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HistoryLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ui/history_line_edit.hpp</header>
   <slots>
    <signal>lineExecuted(QString)</signal>
    <slot>execute()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_send</sender>
   <signal>clicked()</signal>
   <receiver>input_console</receiver>
   <slot>execute()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>699</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>576</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_console</sender>
   <signal>lineExecuted(QString)</signal>
   <receiver>ServerWidget</receiver>
   <slot>rcon_command(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>819</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_reconnect</sender>
   <signal>clicked()</signal>
   <receiver>ServerWidget</receiver>
   <slot>xonotic_reconnect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>508</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>577</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_refresh_status</sender>
   <signal>clicked()</signal>
   <receiver>ServerWidget</receiver>
   <slot>xonotic_request_status()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>588</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_refresh_cvars</sender>
   <signal>clicked()</signal>
   <receiver>ServerWidget</receiver>
   <slot>xonotic_request_cvars()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>561</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>rcon_command(QString)</slot>
  <slot>xonotic_reconnect()</slot>
  <slot>xonotic_request_status()</slot>
  <slot>xonotic_request_cvars()</slot>
 </slots>
</ui>
